src/man/firejail.txt

```diff
@@ -66,7 +66,7 @@ command line options. The default Firejail filesystem is based on the host files
 system directories mounted read-only. These directories are /etc, /var, /usr, /bin, /sbin, /lib, /lib32,
 /libx32 and /lib64. Only /home and /tmp are writable.
 .PP
-As it starts up, Firejail tries to find a security profile based on the name of the application.
+Upon execution Firejail first looks in ~/.config/firejail/ for a profile and if it doesn't find one, it looks in /etc/firejail. For profile resolution detail see https://github.com/netblue30/firejail/wiki/Creating-Profiles#locations-and-types .
```

Comment:
Maybe spilt over multiple lines, very long.

---

If firejail is self-compiled w/o `--prefix=/usr`, profiles are located under `/usr/local/etc` we've often issues where is is one point of confusion. Maybe a it is worth a note.

Reply:
OK, but your additional info about self-compiled w/o `--prefix=/usr` , I would put into wiki not ot clutter man, when there is still wiki.

Reply:
Well I can see that I was confused either. :)

Reply:
@rusty-snake should not be self-compiled w/o `--prefix=/usr` rather somehow mitigated to not happen? Is there any reason to exist? I would be more elegant solution otherwise I should be mentioned everywhere where /etc/firejail/ is metioned. IMO

Reply:
wiki edited 
https://github.com/netblue30/firejail/wiki/Creating-Profiles/8b5f3f2795f67b15f69dae94cb2eeba8afe6457c
https://github.com/netblue30/firejail/wiki/Creating-Profiles/878f36a671b223e126c05ec623d712983138aa17

Reply:
> should not be self-compiled w/o --prefix=/usr rather somehow mitigated to not happen? Is there any reason to exist? It would be more elegant solution otherwise it should be mentioned everywhere where /etc/firejail/ is metioned. IMO

sorry, but I don't get what you are asking.

`--prefix=/usr/local` is default for almost every buildsystem.
There is also the `${CFG}` macro in profiles, however it is so rarely used that it would be more confusing.

Reply:
I am not C programmer thus I do not know about `--prefix`. There must be other contributors like me, who can contribute to .profiles/bugs/documentation. I have installed firajail from sources according to https://github.com/netblue30/firejail#compile-and-install . And now I have /etc/firejail (meybe from previous .deb installation) and /usr/local/etc/firejail. Should not be better to either mention it in https://github.com/netblue30/firejail#compile-and-install or I guess use something like`$ ./configure --prefix=/ && make && sudo make install-strip` to remove confusion? Thx for your time

Reply:
> I am not C programmer 

too

> meybe from previous .deb installation

yes, it is only remove with (`apt purge`) AFAIK

> I guess use something like$ ./configure --prefix=/ && make && sudo make install-strip to remove confusion?

If you install it from source (w/o .deb or .rpm) `./configure --prefix=/usr && make && sudo make install-strip`. `--prefix=/usr` implies `--sysconfdir=/etc` (for firejail).

See [this thread](https://github.com/netblue30/firejail/issues/3333#issuecomment-611623863) for how to install firejail from source.

Reply:
IIRC is `--enable-apparmor` broken w/o `--prefix=/usr`.

Reply:
> And now I have /etc/firejail (meybe from previous .deb installation) and /usr/local/etc/firejail.

Perhaps it's not a bad idea to have proper `uninstall` instructions too. When installed using native OS tools it's simple: use the same tools to uninstall firejail. It potentially gets more complicated with a manual install. Especially if the original install directory no longer exists on the filsystem. In the latter case the conventional `sudo make uninstall` will fail. For that to work succesfully a user will have to redo the installation steps - hopefully remembering the exact ./configure options - prior to running the uninstall command.

This translates to:

(1) redo the manual installation exactly as before
- $ git clone https://github.com/netblue30/firejail.git
- $ cd firejail
- $ ./configure && make && sudo make install-strip

(2) uninstall
- $ sudo make uninstall

@OndrejMalek Tell us if you need assistance with 'unconfusing' your firejail installation.

Reply:
> hopefully remembering the exact ./configure options

hopefully it is in the shellhistory

> sudo make uninstall

In general using `.pkg.tar.xz`, `.deb`, `.rpm` is the prefered way to install software, maybe we should make this way some more prominent.

What do you guys thing about a "using firejail-git" wiki page?
I could maintain a Fedora section, @glitsj16 Arch and Ubuntu?

Reply:
> In general using .pkg.tar.xz, .deb, .rpm is the prefered way to install software, maybe we should make this way some more prominent.

I agree and have made a WIP suggestion on how to achieve this in #3442.

> What do you guys thing about a "using firejail-git" wiki page?
> I could maintain a Fedora section, @glitsj16 Arch and Ubuntu?

Yes that's fine with me. I can always ask @reinerh for assistance on Debian/Ubuntu, although such a firejail-git wiki page shouldn't be that complicated.

Reply:
@glitsj16 we are getting to different issue than this PR is about. I am not competent to solve --prefix thing. Please do it your PR.

But about situation a got into: 
1. On Linux Mint I have installed Firejail from repos  version - 0.9.52-2
2. Some profiles did not work.
3. I have probably unistalled firejail not properly
4. I have installed stable version 0.62.1 apparmor.deb from source forge.
5. Again some unexpected behavior ( https://github.com/netblue30/firejail/issues/3389 )
6. Again I have probably unistalled firejail not properly
7. Installed from sources -  things work right but can not make apparmor work - at this point I did not now about /usr/local/etc/firejail because a had still files in /etc/firejail - so when I was tweaking profiles a was oparating on wrong files at /etc/firejail, which is according to man pages right.
I can see I am quite BFU at this point :D. But this can happen.

Other thing is when I wanna uninstall sources I have to do:
```
sudo firecfg --clean # without this I have to manualy rm links, when I wanna do 'sudo firecfg' next time (probably due to changing --prefix=/usr)
sudo make uninstall
```

Reply:
> things are working right but I can not make apparmor work 

Without `--prefix=/usr`, `firejail-default` (the AA-profile) is installed under /usr/local/etc/apparmor.d. I can't help you with AA, I never actively unsed AA.

Reply:
@rusty-snake thx dont bother with it.  I'll try some other time.